src/thing_creature.c

```diff
@@ -2495,7 +2510,7 @@ void cause_creature_death(struct Thing *thing, CrDeathFlags flags)
             add_creature_to_pool(crmodel, 1, 1);
         thing_death_ice_explosion(thing);
     } else
-    if (shot_model_makes_flesh_explosion(cctrl->shot_model))
+    if ( (shot_model_makes_flesh_explosion(cctrl->shot_model)) || (cctrl->timebomb_death) )
```

Comment:
Would it not be possible to pick this up from the SHOT_TIME_BOMB too?

Reply:
Presumably. What's cctrl->shot_model here?

Reply:
Don't know, but I assumed it would get filled only when a creature gets killed by a shot.